### PR TITLE
tests: added geolocation tests

### PR DESCRIPTION
**Pull request by mxschmitt**

### CODE REVIEW

File: playwright/async_api.py [hunk 1]
```diff
@@ -4922,10 +4924,11 @@ async def setGeolocation(self, geolocation: typing.Dict) -> NoneType:
 
         Parameters
         ----------
-        geolocation : Optional[typing.Dict]
```
Comment (pavelfeldman) [inline playwright/async_api.py:4925]: Let's try restoring parameter docs here?

Comment (mxschmitt) [reply playwright/async_api.py:4925]: good catch. Added a fallback for such cases.

File: playwright/async_api.py [hunk 2]
```diff
@@ -4913,7 +4913,9 @@ async def clearPermissions(self) -> NoneType:
         """
         return mapping.from_maybe_impl(await self._impl_obj.clearPermissions())
 
-    async def setGeolocation(self, geolocation: typing.Dict) -> NoneType:
+    async def setGeolocation(
+        self, latitude: int, longitude: int, accuracy: int = None
+    ) -> NoneType:
         """BrowserContext.setGeolocation
 
         Sets the context's geolocation. Passing `null` or `undefined` emulates position unavailable.
```
Comment (pavelfeldman) [inline playwright/async_api.py:4921]: `Passing null or undefined emulates position unavailable` - how is this working now?

Comment (mxschmitt) [reply playwright/async_api.py:4921]: Probably I have to revert the changes so that its possible again to specify None for the geolocation argument.

Comment (pavelfeldman) [reply playwright/async_api.py:4921]: @dgozman was saying it did not necessarily work due to caching in the browsers, but I'd say staying consistent with upstream is a good idea for now.

**Pull request merged**